File: XMUOJ/GW120/main.py
def move(n:int, a: str, b: str):
    print(n, ":", a, "->", b, sep="")


def hanoi(n: int, a: str, b: str, c: str):
    if n == 1:
        move(1, a, c)
    else:
        hanoi(n - 1, a, c, b)
        move(n, a, c)
        hanoi(n - 1, b, a, c)


n, a, b, c = input().split()
hanoi(int(n), a, b, c)

File: XMUOJ/GW120/main.cpp
#include <bits/stdc++.h>
using namespace std;

void move(int n, char a, char b)
{
	cout << n << ":" << a << "->" << b << endl;
}

void hanoi(int n, char a, char b, char c)
{
	if (n == 1)
		move(1, a, c);
	else
	{
		hanoi(n - 1, a, c, b);
		move(n, a, c);
		hanoi(n - 1, b, a, c);
	}
}

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	int n;
	char a, b, c;
	cin >> n >> a >> b >> c;
	hanoi(n, a, b, c);

	return 0;
}
